samples/gocdk-docstore/main.go

```diff
@@ -0,0 +1,205 @@
+	if err != nil {
+		log.Printf("Failed to open collection: %v\n", err)
+		return subcommands.ExitFailure
+	}
+
+	q := collection.Query()
+	if cmd.date != "" {
+		q = q.Where("Date", "=", cmd.date)
+	}
+	iter := q.Get(ctx)
+	for {
+		var msg Message
+		err := iter.Next(ctx, &msg)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Printf("Failed to list: %v", err)
+			return subcommands.ExitFailure
+		}
+		fmt.Println(msg)
+	}
+	return subcommands.ExitSuccess
+}
+
+type putCmd struct{}
+
+func (*putCmd) Name() string     { return "put" }
+func (*putCmd) Synopsis() string { return "Put an item from stdin" }
+func (*putCmd) Usage() string {
+	return `put <collection URL> <message>
+
+  Read from stdin and put an message with the current timestamp in <collection URL>.
+
+  Example:
+    gocdk-docstore put firestore://mycollection "hello docstore` + helpSuffix
+}
+
+func (*putCmd) SetFlags(_ *flag.FlagSet) {}
+
+func (*putCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	if f.NArg() != 2 {
+		f.Usage()
+		return subcommands.ExitUsageError
+	}
+	collectionURL := f.Arg(0)
+	content := f.Arg(1)
+
+	// Open a *docstore.Collection using the collectionURL.
+	collection, err := docstore.OpenCollection(ctx, collectionURL)
+	if err != nil {
+		log.Printf("Failed to open collection: %v\n", err)
+		return subcommands.ExitFailure
+	}
+
+	t := time.Now()
+	if err := collection.Put(ctx, &Message{
+		Date:      t.Format("2006-01-02"),
+		Timestamp: t.Format(time.RFC3339Nano),
+		Content:   content,
+	}); err != nil {
+		log.Printf("Failed to put message: %v\n", err)
+		return subcommands.ExitFailure
+	}
+	return subcommands.ExitSuccess
+}
+
+type deleteCmd struct {
+	date string
+}
+
+func (*deleteCmd) Name() string     { return "delete" }
+func (*deleteCmd) Synopsis() string { return "Delete items in a collection" }
+func (*deleteCmd) Usage() string {
+	return `delete [-d <date>] <collection URL>
+
+  Delete the docstores in <collection URL>.
+
+  Example:
+    gocdk-docstore delete -d "2006-01-02" firestore://mycollection` + helpSuffix
+}
+
+func (cmd *deleteCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&cmd.date, "d", "", "delete the messages from this date, in the format YYYY-MM-DD")
+}
+
+func (cmd *deleteCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	if f.NArg() != 1 {
+		f.Usage()
+		return subcommands.ExitUsageError
+	}
+	collectionURL := f.Arg(0)
+
+	// Open a *docstore.Collection using the collectionURL.
+	collection, err := docstore.OpenCollection(ctx, collectionURL)
+	if err != nil {
+		log.Printf("Failed to open collection: %v\n", err)
+		return subcommands.ExitFailure
+	}
+
+	q := collection.Query()
+	if cmd.date != "" {
+		q = q.Where("Date", "=", cmd.date)
+	}
+	if err := q.Delete(ctx); err != nil {
+		log.Printf("Failed to delete: %v", err)
+		return subcommands.ExitFailure
+	}
+	return subcommands.ExitSuccess
+}
```

Comment:
Maybe something that shows action lists, like: modify a document (with Update) and then Get and print the new document.

Reply:
Good idea, add an updateCmd.